bgpd/bgp_mplsvpn.c

```diff
@@ -524,46 +524,81 @@ static bool sid_exist(struct bgp *bgp, const struct in6_addr *sid)
  * else: try to allocate as auto-mode
  */
 static uint32_t alloc_new_sid(struct bgp *bgp, uint32_t index,
-			      struct in6_addr *sid_locator,
+			      struct srv6_locator_chunk *sid_locator_chunk,
 			      struct in6_addr *sid)
 {
+	int debug = BGP_DEBUG(vpn, VPN_LEAK_LABEL);
 	struct listnode *node;
 	struct srv6_locator_chunk *chunk;
 	bool alloced = false;
 	int label = 0;
 	uint8_t offset = 0;
-	uint8_t len = 0;
+	uint8_t func_len = 0, shift_len = 0;
+	uint32_t index_max = 0;
 
-	if (!bgp || !sid_locator || !sid)
+	if (!bgp || !sid_locator_chunk || !sid)
 		return false;
 
 	for (ALL_LIST_ELEMENTS_RO(bgp->srv6_locator_chunks, node, chunk)) {
-		*sid_locator = chunk->prefix.prefix;
+		if (chunk->function_bits_length > 20) {
```

Comment:
Can we define somewhere all those magic numbers as macros? 20, 16. Is this like ZEBRA_SRV6_FUNCTION_LENGTH?

Reply:
fixed

* 20: create new constant `BGP_PREFIX_SID_SRV6_MAX_FUNCTION_LENGTH`
* 16: use already-defined constant `MPLS_LABEL_UNRESERVED_MIN`